app/src/main/java/edu/gwu/androidtweets/TweetsActivity.kt

```diff
@@ -10,12 +10,68 @@ import androidx.appcompat.app.AppCompatActivity
 import androidx.core.view.isVisible
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.tasks.Tasks
 import com.google.android.material.floatingactionbutton.FloatingActionButton
 import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.database.*
 import org.jetbrains.anko.doAsync
+import java.util.*
+import kotlin.collections.ArrayList
 import kotlin.random.Random
 
+object UserInfo
+
+object FetchFriendsListExample {
+
+    val firebaseDb = FirebaseDatabase.getInstance()
+
+    val userInfo = mutableListOf<UserInfo>()
+
+    fun fetchUserInfoList(userIds: List<String>, onCompleteListener: (List<UserInfo>) -> Unit) {
+        val reference = "users"
+        val userIdsQueue: Queue<String> = LinkedList(userIds)
+
+        val valueEventListener = object : ValueEventListener {
+            override fun onCancelled(error: DatabaseError) {
+                // ...
+            }
+
+            override fun onDataChange(snapshot: DataSnapshot) {
+                // Parse out user data from snapshot
+                userInfo.add(...)
+
+                // If there are more users we need to fetch, continue getting user info
+                val nextId = userIdsQueue.poll()
+                if (nextId != null) {
+                    firebaseDb.getReference("users/$nextId").addListenerForSingleValueEvent(this)
+                } else {
+                    // Collection is finished, invoke the original caller with the complete list
+                    onCompleteListener(userInfo)
+                }
+            }
+        }
+
+        // addListenerForSingleValueEvent is the same as addValueEventListener, except it "unsubscribes"
+        // from updates after the first data event
+        // See: https://firebase.google.com/docs/database/android/read-and-write#read_data_once
+        val firstId = userIdsQueue.poll()
+        firebaseDb.getReference("users/$firstId").addListenerForSingleValueEvent(valueEventListener)
+
+
+        // .get() gives you a "pending data Task" which you can block and wait on
+        // See: https://firebase.google.com/docs/database/android/read-and-write#read_data_once
+        val pendingData: Task<DataSnapshot> = firebaseDb.getReference("users/...").get()
+
+        // Blocks the thread until you have a result
+        // Now you need to handle all of the threading and error handling (try-catch) similar to what we've done with API calls
+        // Also see: https://developers.google.com/android/guides/tasks#chaining
+        val result: DataSnapshot = Tasks.await(pendingData)
+
+    }
+
+}
```

Comment:
Accidentally committed this - but this is testing out some different ways to pull data from Firebase DB :)